### PR TITLE
Add flake8

### PR DESCRIPTION
**Pull request by nileshpatra**
The coding style followed is not consistent in the codebase.
Added in flake8 tests so that the new code added would be consistent.
Also fixed the current failing tests, and overridden redundant once.

### CODE REVIEW

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5058765&bid=16650535)

**Comment by nileshpatra** [comment]
To test locally:
`flake8 . --count --exit-zero --max-complexity=10 --max-line-length=500 --statistics --ignore=C901,E251,E722,E231,E902 --exclude=.git,.venv,.gitignore`

**Comment by nileshpatra** [comment]
@rpotter12 please take a review^

File: wplay/onlinetracker.py [hunk 1]
```diff
@@ -1,23 +1,19 @@
...
-    target_name = await target_search.search_and_select_target(page, target, hide_groups=True)
-
-    Path(data_folder_path/'tracking_data').mkdir(parents=True, exist_ok=True)
-
-    status_file = open(data_folder_path/'tracking_data'/f'status_{target_name}.txt', 'w').close()
-    status_file = open(data_folder_path/'tracking_data'/f'status_{target_name}.txt', 'a')
+    target_name = await target_search.search_and_select_target(page, target, hide_groups = True)
+    os.path.join(data_folder_path, 'tracking_data').mkdir(parents = True, exist_ok = True)
+    status_file = open(os.path.join(data_folder_path, 'tracking_data', f'status_{target_name}.txt'), 'w').close()
+    status_file = open(os.path.join(data_folder_path, 'tracking_data', f'status_{target_name}.txt'), 'a')
```
Comment (xandao6) [inline wplay/onlinetracker.py]: Why did you change pathlib to os? pathlib is the right way to deal with paths in python3 +. 
Take a look here. [PEP428](https://www.python.org/dev/peps/pep-0428/)

File: wplay/utils/helpers.py [hunk 2]
```diff
@@ -27,9 +27,9 @@
 # endregion
 
 # region PATHS
-data_folder_path = Path.home()/'wplay'
-logs_path = Path.home()/'wplay'/'logs'
-user_data_folder_path = Path.home()/'wplay'/'.userData'
+data_folder_path = os.path.join(Path.home(), 'wplay')
+logs_path = os.path.join(Path.home(), 'wplay', 'logs')
+user_data_folder_path = os.path.join(Path.home(), 'wplay', '.userData')
```
Comment (xandao6) [inline wplay/utils/helpers.py]: Same here, is much easier: `Path.home()/'wplay'` than `os.path.join(Path.home(), 'wplay')`

File: wplay/savechat.py [hunk 3]
```diff
@@ -108,15 +105,15 @@ def jsonPrint(data):
 
 
 def localFileLog(md5):
-    logfile = 'logs'+os.path.sep+'files.log'
+    logfile = 'logs' + os.path.sep + 'files.log'
```
Comment (xandao6) [inline wplay/savechat.py]: `'logs' + os.path.sep + 'files.log'` this could be : `'logs'/'files.log'`

**Comment by xandao6** [comment]
We could remove almost all from os and move to pathlib.

**Comment by xandao6** [comment]
Add flake8 to setup requirements too!

**Comment by nileshpatra** [comment]
> We could remove almost all from os and move to pathlib.

I'll try doing this

> Add flake8 to setup requirements too!

I have already added this, else circleci would have failed - check commits

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5058765&bid=16650535)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5058765&bid=16650535)

**Comment by nileshpatra** [comment]
@xandao6 done. Please review?

File: wplay/utils/browser_config.py [hunk 4]
```diff
@@ -72,15 +66,15 @@ def __init__(self, *args: Any, **kwargs: Any) -> None:
...
-async def __config_browser(username=None, save_session=False):
+async def __config_browser(username = None, save_session = False):
     if username is not None and username != '' and save_session:
         return await launch(
-            headless=False,
-            autoClose=False,
-            userDataDir=user_data_folder_path/username
+            headless = False,
+            autoClose = False,
+            userDataDir = "user_data_folder_path" + "/" + "username"
```
Comment (xandao6) [inline wplay/utils/browser_config.py]: `userDataDir = "user_data_folder_path" / "username"`

File: wplay/tgbot.py [hunk 5]
```diff
@@ -17,48 +15,48 @@ def start_tkinter():
 def ask_where_are_the_status_file():
     print('Choose a status text file.')
     status_file_path = filedialog.askopenfile(
-        initialdir = data_folder_path/'tracking_data/',
-        title='Choose a status text file.',
-        filetypes=(("text files","*.txt"),("all files","*.*"))
+        initialdir = os.path.join(data_folder_path, 'tracking_data/'),
```
Comment (xandao6) [inline wplay/tgbot.py]: data_folder_path / 'tracking_data'

File: wplay/tgbot.py [hunk 6]
```diff
@@ -1,9 +1,7 @@
 import tkinter
+import os
```
Comment (xandao6) [inline wplay/tgbot.py]: Check, I think you can remove it

File: wplay/onlinetracker.py [hunk 7]
```diff
@@ -40,15 +36,12 @@
...
-                    f'{datetime.now().strftime("%d/%m/%Y, %H:%M:%S")}' +
-                    f' - Status: {status}\n')
+                    f'{datetime.now().strftime("%d/%m/%Y, %H:%M:%S")}' + f' - Status: {status}\n')
             last_status = is_online
             time.sleep(0.5)
     finally:
         status_file.close()
-        print(f'\nStatus file saved in: ' +
-                str(data_folder_path/'tracking_data'/f'status_{target_name}.txt'))
+        print(f'\nStatus file saved in: ' + str(data_folder_path + '/tracking_data' + '/' + f'status_{target_name}.txt'))
```
Comment (xandao6) [inline wplay/onlinetracker.py]: `print(f'\nStatus file saved in: ' + str(data_folder_path / 'tracking_data' / f'status_{target_name}.txt'))`

File: wplay/utils/target_search.py [hunk 8]
```diff
@@ -127,21 +127,16 @@
...
 
+
 # contact_list_unchecked is a zip (list of tuples) of contact_titles and
 # contact elements, unchecked.
-def __zip_contact_titles_and_elements_unchecked(
-    contact_titles_unchecked,
-    contact_list_elements_unchecked):
-    contact_list_unchecked = list(zip(
-        contact_titles_unchecked, contact_list_elements_unchecked))
+def __zip_contact_titles_and_elements_unchecked(contact_titles_unchecked, contact_list_elements_unchecked):
```
Comment (xandao6) [inline wplay/utils/target_search.py]: https://www.python.org/dev/peps/pep-0008/#indentation

File: wplay/utils/target_search.py [hunk 9]
```diff
@@ -127,21 +127,16 @@
...
+    contact_list_unchecked = list(zip(contact_titles_unchecked, contact_list_elements_unchecked))
     return contact_list_unchecked
 
 
-def __zip_group_titles_and_elements_unchecked(
-    group_titles_unchecked,
-    group_list_elements_unchecked):
-    group_list_unchecked = list(
-        zip(group_titles_unchecked, group_list_elements_unchecked))
+def __zip_group_titles_and_elements_unchecked(group_titles_unchecked, group_list_elements_unchecked):
```
Comment (xandao6) [inline wplay/utils/target_search.py]: https://www.python.org/dev/peps/pep-0008/#indentation

File: wplay/utils/target_data.py [hunk 10]
```diff
@@ -6,10 +6,10 @@
 
 # region FOR SCRIPTING
 async def get_last_seen_from_focused_target(page):
-    #await page.waitForSelector(whatsapp_selectors_dict['status'], visible = True)
+    # Await page.waitForSelector(whatsapp_selectors_dict['status'], visible = True)
```
Comment (xandao6) [inline wplay/utils/target_data.py]: await

File: wplay/utils/session_manager.py [hunk 11]
```diff
@@ -109,38 +109,38 @@ def __verify_answers(answers_menu, data_filenames, question_overwrite):
...
+
     return username, save_session
 
 
 def __verify_if_session_file_exists(data_filenames, username, question_overwrite):
     if username in data_filenames:
         answer_overwrite = prompt(question_overwrite, style=menu_style)
         if answer_overwrite['overwrite_data']:
-            __delete_session_data(user_data_folder_path/username)
+            __delete_session_data(str(user_data_folder_path + '/' + username))
```
Comment (xandao6) [inline wplay/utils/session_manager.py]: `user_data_folder_path / username` is different from `str(user_data_folder_path + '/' + username)`

File: wplay/utils/session_manager.py [hunk 12]
```diff
@@ -109,38 +109,38 @@ def __verify_answers(answers_menu, data_filenames, question_overwrite):
...
     # Handle when person choose 'Continue without saving'
     elif answers_menu['user_options'] == user_options['continue']:
-        username = None; save_session = False
+        username, save_session = None, False
 
     # Handle when person choose 'Delete a session'
     elif answers_menu['user_options'] == user_options['delete']:
         if len(answers_menu['delete']) > 0:
-            [__delete_session_data(user_data_folder_path/username) for username in answers_menu['delete']]
+            [__delete_session_data(str(user_data_folder_path + '/' + username)) for username in answers_menu['delete']]
```
Comment (xandao6) [inline wplay/utils/session_manager.py]: `[__delete_session_data(user_data_folder_path / username) for username in answers_menu['delete']]`

File: wplay/savechat.py [hunk 13]
```diff
@@ -141,36 +138,35 @@ def getMultipleFiles(data, folder):
     files = localFileList()
     data = json.loads(data)
     for entries in data:
-        if any(entries['m'] in lists for lists in files) == \
+        if any(entries['m'] in lists for lists in files) is \
                 False or 'database' in entries['f'].lower():
-            local = folder+os.path.sep+entries['f'].replace("/", os.path.sep)
+            local = folder + '/' + entries['f'].replace("/", os.path.sep)
```
Comment (xandao6) [inline wplay/savechat.py]: `local = folder+os.path.sep+entries['f'].replace("/", os.path.sep)`

File: wplay/savechat.py [hunk 14]
```diff
@@ -108,15 +105,15 @@ def jsonPrint(data):
...
     if not os.path.exists(os.path.dirname(logfile)):
         os.makedirs(os.path.dirname(logfile))
     with open(logfile, 'a') as log:
-        log.write(md5+'\n')
+        log.write(md5 + '\n')
 
 
 def localFileList():
-    logfile = 'logs'+os.path.sep+'files.log'
+    logfile = 'logs' + '/' + 'files.log'
```
Comment (xandao6) [inline wplay/savechat.py]: logfile = 'logs' / 'files.log'

**Comment by xandao6** [comment]
Click in resolver conversation when done.

**Comment by rpotter12** [comment]
@nileshpatra update your fork too. Your fork is 2 commits behind. :)

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 15
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5058765&bid=16650535)

**Comment by nileshpatra** [comment]
@rpotter12 probably because you merged the other PR meanwhile, rebased and pushed, please take a look.
@xandao6 review please?
